connectivity/common.c

```diff
@@ -0,0 +1,77 @@
+#include <regex.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "cJSON.h"
+#include "common.h"
+#include "common/logger.h"
+#include "common/ta_errors.h"
+
+#define CONN_LOGGER "connectivity"
+static logger_id_t logger_id;
+
+void conn_logger_init() { logger_id = logger_helper_enable(CONN_LOGGER, LOGGER_DEBUG, true); }
+
+int conn_logger_release() {
+  if (logger_helper_destroy() != RC_OK) {
+    ta_log_error("Destroying logger failed %s.\n", CONN_LOGGER);
+    return EXIT_FAILURE;
+  }
+
+  return 0;
+}
+
+status_t api_path_matcher(char const *const url, char *const regex_rule) {
+  if (regex_rule == NULL) {
+    ta_log_error("%s\n", "SC_HTTP_NULL");
+    return SC_HTTP_NULL;
+  }
+  regex_t reg;
+  regmatch_t pmatch;
+  status_t ret = SC_OK;
+  int reg_flag = REG_EXTENDED;
+
+  if (regcomp(&reg, regex_rule, reg_flag) != 0) {
+    ta_log_error("%s\n", "SC_HTTP_INVALID_REGEX");
+    return SC_HTTP_INVALID_REGEX;
+  }
+  if (regexec(&reg, url, 1, &pmatch, 0) != 0) {
+    // Did not match pattern
+    ret = SC_HTTP_URL_NOT_MATCH;
+  } else {
+    if ((size_t)(pmatch.rm_eo - pmatch.rm_so) != strlen(url)) {
+      ret = SC_HTTP_URL_NOT_MATCH;
+    }
+  }
+
+  regfree(&reg);
+  return ret;
+}
+
+status_t set_response_content(status_t ret, char **json_result) {
```

Comment:
You do merge two `set_response_content` in `MQTT` and `HTTP` to a common one and change the behavior what original `set_response_conetent` do. 
That is, remove the JSON related code which creates key-value pairs and print to the `json_result`.

_Does_ this modification affect the HTTP connection?

Reply:
Since those response messages are fixed strings, I think we can define the JSON-format string outside (the defined strings are in `connectivity/common.h` now) rather than creating cJSON object and delete them every time.

This modification will not affect the original `set_response_content` functionality.